internal/resources/import_filter.go

```diff
@@ -50,58 +56,259 @@ func (r *ImportFilter) Configure(ctx context.Context, req resource.ConfigureRequ
 		)
 		return
 	}
-	r.Data = data
+	f.Data = data
+}
+
+// fetchImportFilter fetches the import filter from the API endpoint.
+// returns the import filter function as a string.
+func (f *ImportFilter) fetchImportFilter(ctx context.Context, organizationId, projectId, clusterId, appServiceId, keyspace string) (string, error) {
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/importFilter",
+		f.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		keyspace,
+	)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := f.Client.ExecuteWithRetry(ctx, cfg, nil, f.Token, nil)
+	if err != nil {
+		return "", err
+	}
+	return string(response.Body), nil
 }
 
 // Create upserts the import filter.
-func (r *ImportFilter) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+func (f *ImportFilter) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var plan providerschema.ImportFilter
 	diags := req.Plan.Get(ctx, &plan)
 	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	// TODO (AV-104546) - Implement API call to putImportFilter and set state.
+	var (
+		organizationId = plan.OrganizationId.ValueString()
+		projectId      = plan.ProjectId.ValueString()
+		clusterId      = plan.ClusterId.ValueString()
+		appServiceId   = plan.AppServiceId.ValueString()
+		keyspace       = fmt.Sprintf(
+			"%s.%s.%s",
+			plan.AppEndpointName.ValueString(),
+			plan.Scope.ValueString(),
+			plan.Collection.ValueString(),
+		)
+	)
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s/importFilter",
+		f.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		appServiceId,
+		keyspace,
+	)
+
+	headers := map[string]string{"Content-Type": "application/javascript"}
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusNoContent}
+	_, err := f.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		plan.ImportFilter.ValueString(),
+		f.Token,
+		headers,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Creating Import Filter",
+			"Could not upsert Import Filter: "+api.ParseError(err),
+		)
+		return
+	}
+
+	// Refresh from app service to ensure state matches remote
+	body, err := f.fetchImportFilter(ctx, organizationId, projectId, clusterId, appServiceId, keyspace)
+	if err != nil {
+		resp.Diagnostics.AddWarning(
+			"Error reading Import Filter after creation",
+			api.ParseError(err),
```

Comment:
refresh is typically used for retrieving computed attributes.

`Read` handler is used to detect differences in the resource.